core/DataTable/Filter/ColumnCallbackReplace.php

```diff
@@ -99,6 +99,11 @@ public function filter($table)
                 $this->filterSubTable($row);
             }
         }
+
+        if (in_array('label', $this->columnsToFilter)) {
+            // we need to force rebuilding the index
+            $table->setLabelsHaveChanged();
```

Comment:
If we change the label column, we need to make sure the dataTableLabelToRowIdIndex will be updated. Otherwise `getRowFromLabel()` won't work.